core/modules/admin_bar/admin_bar.module

```diff
@@ -244,6 +244,22 @@ function admin_bar_preprocess_page(&$variables) {
     $settings['position_fixed'] = $setting;
   }
 
+  // Send whether this is an admin page to JS for back to site functionality.
+  $settings['current_path_is_admin'] = FALSE;
+  $current_path = current_path();
+  if (path_is_admin($current_path)) {
+    $settings['current_path_is_admin'] = TRUE;
+  }
+  $alias = backdrop_get_path_alias($current_path, $language->langcode);
+  $settings['current_path'] = $alias;
+
+  $components = $config->get('components');
+  if (in_array('admin_bar.back_to_site_link', $components) ) {
```

Comment:
See my comment above in `admin_bar.inc`. If we decide to go the route of what I explained there, then this line should be:

```php
if (in_array('admin_bar.icon', $components) && in_array('admin_bar.back_to_site_link', $components)) {
```

Also, check the unnecessary space between parentheses at the end.